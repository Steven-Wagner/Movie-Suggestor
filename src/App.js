import React, { Component } from 'react';
import LandingPage from './landing-page/landing-page'
import { Route } from 'react-router-dom';
import Signup from './sign-up/sign-up';
import {movieSuggestions, users} from './dataModel/fakeData'
import Homepage from './home-page/home-page';
import FriendSuggester from './friend-suggestions/friend-suggestions';
import ReviewMovie from './review-movie/review-movie';
import Login from './login/login';

class App extends Component {
  render() {
    return (
      <main className="App">
        <Route 
          exact path="/"
          component={LandingPage}
        />
        <Route 
          path="/signup"
          render={({history})=> {
            return <Signup
            clickCancel={()=> history.push('/')}
            />
          }}
        />
        <Route 
          path="/newreview"
          component={ReviewMovie}
        />
        <Route 
          path="/login"
          render={({history})=>
            <Login clickCancel={()=> history.push('/')}
            />
          }
        />
        <Route 
          path="/homepage"
          render={()=>
            <Homepage movieSuggestions={movieSuggestions}/>
          }
        />
        <Route 
          path="/friendsuggestions"
          render={()=>
            <FriendSuggester users={users}/>
          }
        />
      </main>
    );
  }
}

export default App;
