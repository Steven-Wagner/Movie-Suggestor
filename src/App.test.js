import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import App from './App';

describe('<App/>', () => {
    it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
        <Router>
            <App />
        </Router>,
        div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(
            <Router>
                <App/>
            </Router>)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
})