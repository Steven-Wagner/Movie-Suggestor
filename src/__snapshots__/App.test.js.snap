// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders the UI as expected 1`] = `
<main
  className="main-app-container remote-edge"
>
  <div>
    <div
      className="nav-wrapper"
    >
      <nav>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="nav-button remote-button"
          >
            Sign-up
          </button>
        </a>
        <a
          href="/login"
          onClick={[Function]}
        >
          <button
            className="nav-button remote-button"
          >
            Login
          </button>
        </a>
      </nav>
    </div>
    <main
      role="main"
    >
      <header
        className="main-banner"
        role="banner"
      >
        <h1>
          <span
            className="fas fa-film"
          />
          Movie Suggestor
        </h1>
        <h2
          className="subtitle"
        >
          Personalized Movie Suggestions
        </h2>
      </header>
      <section
        className="remote-edge"
      >
        <header
          className="landing-page-headers"
        >
          <h3>
            Get suggestions from people just like you
          </h3>
        </header>
        <p>
          [
          <em>
            placeholder for screenshot movie suggestions
          </em>
          ]
        </p>
        <p>
          Movie Suggestor allows you to follow people that have similar movie intrests and gives you movie suggestions based on the friends you follow.
        </p>
      </section>
      <section
        className="remote-edge"
      >
        <header
          className="landing-page-headers"
        >
          <h3>
            Review Movies
          </h3>
        </header>
        <p>
          [
          <em>
            placeholder for screenshot of movie review form
          </em>
          ]
        </p>
        <p>
          Review movies in seconds and help the people that follow you get personalized movie suggestions.
        </p>
      </section>
      <section
        className="remote-edge"
      >
        <header
          className="landing-page-headers"
        >
          <h3>
            Start Getting Suggestions Now
          </h3>
          <a
            href="/signup"
            onClick={[Function]}
          >
            Sign-up!
          </a>
        </header>
      </section>
    </main>
    <footer
      className="remote-edge"
    >
      Built by Steven Wagner. Movie data from 
      <a
        href="OMDB.com"
        target="_blank"
      >
        omdbapi.com
      </a>
      . 2019
    </footer>
  </div>
</main>
`;
